### PR TITLE
24 mwmealcleaner

### PR DESCRIPTION
**Pull request by Kangaroo6942**
added a cleaner for the Meals that takes the Meals from the Parser and changes them according to the desired standards. Happy about feedback on names and things I missed

### CODE REVIEW

File: src/MensaWidget/MWMealCleaner.class.st [hunk 1]
```diff
@@ -0,0 +1,81 @@
+"
+A MWMealCleaner is a FIlter for Meal lists provided by the parsing after processing the JSON Strring.
```
Comment (lchristus) [inline src/MensaWidget/MWMealCleaner.class.st:2]: one to many r's in string

File: src/MensaWidget/MWMealCleaner.class.st [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+MWMealCleaner class >> fixMeal: aMeal [
+
+	| meal |
+	
+	meal := aMeal.
+	meal isSaladBar ifTrue: [meal := MWMeal saladBar].
+	meal isHotCounter ifTrue: [meal := MWMeal hotCounter].
+	self fixPrices: meal.
+	self fixCategories: meal.
```
Comment (lchristus) [inline src/MensaWidget/MWMealCleaner.class.st:57]: cascade

**Review by lchristus — CHANGES_REQUESTED** [review]
Overall very nice but we still need to make sure that all prices have two places after the decimal point tha ti think would still be good to have

File: src/MensaWidget/MWMealCleaner.class.st (hunk 2 above)
Comment (Kangaroo6942) [reply src/MensaWidget/MWMealCleaner.class.st]: done

File: src/MensaWidget/MWMealCleaner.class.st (hunk 1 above)
Comment (Kangaroo6942) [reply src/MensaWidget/MWMealCleaner.class.st]: done

**Comment by Kangaroo6942** [comment]
i have now implemented the floats to always have two decimals after the dot

**Review by OscarUlb — APPROVED** [review]
Very good and clean! Thanks! My only question is regarding the MW Price object. I’m not sure how necessary it is and how well it fits with the simple design. But I am fine with keeping it.

**Comment by Kangaroo6942** [comment]
I totally see that, i'd like someone else's feedback regarding this topic. If there is another opinion against the Price object I will use the JSON

File: src/MensaWidget/MWMealCleaner.class.st [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+}
+MWMealCleaner class >> fixPrices: aMeal [
+	
+	| price |
+	
+	aMeal prices ifNil: [aMeal prices: self emptyPrice. 
+		^ nil].
+	price := aMeal prices.
+	"aMeal prices class = MWPrice ifTrue: [^ nil].
+	price := (MWParser put: '[', aMeal prices asJsonString, ']' in: MWPrice) first."
```
Comment (OscarUlb) [inline src/MensaWidget/MWMealCleaner.class.st:96]: Is this comment intentional or a debugging leftover?

Comment (Kangaroo6942) [reply src/MensaWidget/MWMealCleaner.class.st]: done

**Review by lchristus — APPROVED** [review]

File: src/MensaWidget/MWDataGettingWrapper.class.st [hunk 4]
```diff
@@ -35,7 +35,7 @@ MWDataGettingWrapper class >> getMealsOnDate: aDate [
 	canteenDict := self canteenDict.
 	canteenDict keys do: [ :id |
```
[the remark this answers is not in the document]
Comment (Kangaroo6942) [reply src/MensaWidget/MWDataGettingWrapper.class.st:36]: that is correct, but this class is actually not the scope of this ticket. There is an extra unify spaces ticket where all classes will be adapted.

File: src/MensaWidget/MWDataGettingWrapper.class.st [hunk 5]
```diff
@@ -35,7 +35,7 @@ MWDataGettingWrapper class >> getMealsOnDate: aDate [
...
 		json := MWApiWrapper getMealsFrom: id on: aDate.
-		canteens add: (MWCanteen newWithId: id  Name: (canteenDict at: id) Date: aDate Meals: (MWParser put: json in: MWMeal)) ].
```
[the remark this answers is not in the document]
Comment (Kangaroo6942) [reply src/MensaWidget/MWDataGettingWrapper.class.st:38]: as stated above

File: src/MensaWidget/MWDataGettingWrapper.class.st (hunk 4 above)
[the remark this answers is not in the document]
Comment (tommccaughey) [reply src/MensaWidget/MWDataGettingWrapper.class.st:36]: I see, grand!

**Pull request merged**